ui/js/directives/treeherder/clonejobs.js

```diff
@@ -526,29 +549,33 @@ treeherder.directive('thCloneJobs', [
 
             //Empty the job column before populating it
             jobTdEl.empty();
-
-            var jgObj, jobGroup, i;
-            for (i=0; i<jobGroups.length; i++) {
-
-                jgObj = jobGroups[i];
-
+            var jgObj, jobTdHtml = "";
+            jobGroups.forEach(function (jobGroup) {
+                jgObj = jobGroup;
                 if (jgObj.symbol !== '?') {
                     // Job group detected, add job group symbols
-                    jobGroups[i].grkey = jgObj.mapKey;
-                    jobGroups[i].collapsed = true;
-                    jobGroup = $(jobGroupInterpolator(jobGroups[i]));
-                    jobTdEl.append(jobGroup);
-
+                    jobGroup.grkey = jgObj.mapKey;
+                    jobGroup.collapsed = true;
                     if (isGroupExpanded(jgObj)) {
-                        addJobBtnEls(jgObj, jobGroup.find(".group-job-list"));
+                        jobGroup.btnHTML = getJobBtnEls(jgObj);
                     } else {
-                        addGroupJobsAndCounts(jgObj, jobGroup);
+                        var btnArrays = renderGroupJobsAndCounts(jgObj);
+                        jobGroup.btnHTML = "";
+                        btnArrays.jobBtnArray.forEach(function(element) {
+                            jobGroup.btnHTML += element;
+                        });
+                        jobGroup.countBtnHTML = "";
+                        btnArrays.jobCountBtnArray.forEach(function(element) {
+                            jobGroup.countBtnHTML += element;
+                        });
                     }
+                    jobTdHtml += jobGroupInterpolator(jobGroup);
                 } else {
                     // Add the job btn spans
-                    addJobBtnEls(jgObj, jobTdEl);
+                    jobTdHtml += getJobBtnEls(jgObj);
                 }
-            }
+            });
+            jobTdEl.append($(jobTdHtml));
```

Comment:
I think we can replace this with `jobTdEl.html(jobTdHtml)`: http://api.jquery.com/html/#html2

That would make the `jobTdEl.empty()`, above, unnecessary.